diag_manager/fms_diag_axis_object.F90

```diff
@@ -0,0 +1,287 @@
+     CHARACTER(len=:),   ALLOCATABLE, private :: units           !< Units of the axis
+     CHARACTER(len=:),   ALLOCATABLE, private :: long_name       !< Long_name attribute of the axis
+     CHARACTER(len=1)               , private :: cart_name       !< Cartesian name "X", "Y", "Z", "T", "U", "N"
+     CLASS(*),           ALLOCATABLE, private :: axis_data(:)    !< Data of the axis
+     !< TO DO this can be a dlinked to avoid having limits
+     type(subaxis_t)                , private :: subaxis(3)      !< Array of subaxis
+     integer                        , private :: nsubaxis        !< Number of subaxis
+     class(fms_domain_t),ALLOCATABLE, private :: axis_domain     !< Domain
+     INTEGER                        , private :: length          !< Global axis length
+     INTEGER                        , private :: direction       !< Direction of the axis 0, 1, -1
+     INTEGER                        , private :: edges           !< Axis ID for the previously defined "edges axis"
+     CHARACTER(len=128)             , private :: aux             !< Auxiliary name, can only be <TT>geolon_t</TT>
+                                                                 !! or <TT>geolat_t</TT>
+     CHARACTER(len=128)             , private :: req             !< Required field names.
+     INTEGER                        , private :: tile_count      !< The number of tiles
+     TYPE(diag_atttype),allocatable , private :: attributes(:)   !< Array to hold user definable attributes
+     INTEGER                        , private :: num_attributes  !< Number of defined attibutes
+     INTEGER                        , private :: domain_position !< The position in the doman (NORTH or EAST or CENTER)
+
+     contains
+
+     PROCEDURE :: register => diag_axis_init
+     PROCEDURE :: axis_length => get_axis_length
+     PROCEDURE :: set_subaxis
+
+     ! TO DO:
+     ! PROCEDURE :: write_axis_metadata
+     ! PROCEDURE :: write_axis_data
+     ! PROCEDURE :: get_fileobj_type_needed (use the domain to figure out what fms2 fileobj to use)
+     ! Get/has/is subroutines as needed
+  END TYPE diag_axis_t
+
+  !> @addtogroup fms_diag_yaml_mod
+  !> @{
+  contains
+
+  !!!!!!!!!!!!!!!!! DIAG AXIS PROCEDURES !!!!!!!!!!!!!!!!!
+  !> @brief Initialize the axis
+  subroutine diag_axis_init(obj, name, axis_data, units, cart_name, long_name, direction,&
+  & set_name, edges, Domain, Domain2, DomainU, aux, req, tile_count, domain_position )
+    class(diag_axis_t)                       :: obj             !< Diag_axis obj
+    CHARACTER(len=*),   INTENT(in)           :: name            !< Name of the axis
+    class(*),           INTENT(in)           :: axis_data(:)    !< Array of coordinate values
+    CHARACTER(len=*),   INTENT(in)           :: units           !< Units for the axis
+    CHARACTER(len=*),   INTENT(in)           :: cart_name       !< Cartesian axis ("X", "Y", "Z", "T", "U", "N")
```

Comment:
The `len` for `cart_name` should always be 1.  Is there any reason to not have it be 1?